File: januswrapper/src/main/java/com/clearkeep/januswrapper/common/AvatarImageTask.java
package com.clearkeep.januswrapper.common;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.AsyncTask;
import android.widget.ImageView;
import java.io.InputStream;
import java.lang.ref.WeakReference;
import java.net.URL;
import de.hdodenhof.circleimageview.CircleImageView;

public class AvatarImageTask extends AsyncTask<String, Void, Bitmap> {

    WeakReference<CircleImageView> avatarViewWef;

    public AvatarImageTask(CircleImageView avatarView) {
        this.avatarViewWef = new WeakReference<>(avatarView);
    }

    protected Bitmap doInBackground(String... urls) {
        String avatarUrl = urls[0];
        Bitmap bmp = null;
        try {
            InputStream in = new URL(avatarUrl).openStream();
            bmp = BitmapFactory.decodeStream(in);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return bmp;
    }
    protected void onPostExecute(Bitmap result) {
        ImageView img = avatarViewWef.get();
        if (img != null && result != null) {
            img.setImageBitmap(result);
        }
    }
}

File: app/src/main/java/com/clearkeep/presentation/screen/videojanus/common/AvatarImageTask.kt
package com.clearkeep.presentation.screen.videojanus.common

import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.os.AsyncTask
import android.widget.ImageView
import de.hdodenhof.circleimageview.CircleImageView
import java.lang.ref.WeakReference
import java.net.URL

class AvatarImageTask(avatarView: CircleImageView) : AsyncTask<String?, Void?, Bitmap?>() {
    var avatarViewWef: WeakReference<CircleImageView> = WeakReference(avatarView)

    override fun doInBackground(vararg urls: String?): Bitmap? {
        val avatarUrl = urls[0]
        var bmp: Bitmap? = null
        try {
            val `in` = URL(avatarUrl).openStream()
            bmp = BitmapFactory.decodeStream(`in`)
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return bmp
    }

    override fun onPostExecute(result: Bitmap?) {
        val img: ImageView? = avatarViewWef.get()
        if (img != null && result != null) {
            img.setImageBitmap(result)
        }
    }
}